### PR TITLE
Misc refactorings and fixes to shutdown sequence

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
This PR contains misc refactorings extracted from a branch with changes I've been working on (PR coming soon), including some fixes to the shutdown sequence.

Two user-facing changes:

1. Fixed: cannot stop daprd if it's waiting for the app to come online (often happens if the app crashed while using the Dapr CLI)
2. Can force shutdown (aborting any graceful shutdown sequence) by sending a second SIGTERM/SIGINT.

### CODE REVIEW

File: pkg/grpc/dial.go [hunk 1]
```diff
@@ -1,18 +1,28 @@
+//go:build !windows
```
Comment (ItalyPaleAle) [inline pkg/grpc/dial.go:1]: Reviewers: using build tags here rather than runtime checks

File: pkg/grpc/api.go [hunk 2]
```diff
@@ -68,52 +68,17 @@ const (
 )
 
 // API is the gRPC interface for the Dapr gRPC API. It implements both the internal and external proto definitions.
-//
-//nolint:interfacebloat
 type API interface {
 	// DaprInternal Service methods
-	CallActor(ctx context.Context, in *internalv1pb.InternalInvokeRequest) (*internalv1pb.InternalInvokeResponse, error)
-	CallLocal(ctx context.Context, in *internalv1pb.InternalInvokeRequest) (*internalv1pb.InternalInvokeResponse, error)
+	internalv1pb.ServiceInvocationServer
```
Comment (ItalyPaleAle) [inline pkg/grpc/api.go:73]: Reviewers: by using the interface from the protos, we can simplify this a lot (and make it easier on us when we change protos)

File: pkg/channel/grpc/grpc_channel.go [hunk 3]
```diff
@@ -150,9 +151,20 @@ func (g *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
...
+var emptyPbPool = sync.Pool{
+	New: func() any {
+		return &emptypb.Empty{}
+	},
+}
+
 // HealthProbe performs a health probe.
 func (g *Channel) HealthProbe(ctx context.Context) (bool, error) {
-	_, err := g.appHealthClient.HealthCheck(ctx, &emptypb.Empty{})
+	// We use the low-level method here so we can avoid allocating multiple &emptypb.Empty and use the pool
```
Comment (ItalyPaleAle) [inline pkg/channel/grpc/grpc_channel.go:162]: This change should allow us to be a bit easier on the GC here.

File: pkg/runtime/cli.go [hunk 4]
```diff
@@ -228,9 +228,18 @@ func FromFlags() (*DaprRuntime, error) {
 		concurrency = *appMaxConcurrency
 	}
 
-	appPrtcl := string(HTTPProtocol)
-	if *appProtocol != string(HTTPProtocol) {
-		appPrtcl = *appProtocol
+	var appProtocol string
```
Comment (ItalyPaleAle) [inline pkg/runtime/cli.go:231]: Note this changes actually enforces protocol to be one of "http" or "grpc" (case-insensitive)

**Review by berndverst — APPROVED** [review]

**Review by artursouza — APPROVED** [review]
Can we have standalone E2E test as a follow up?

**Pull request merged**